app/producers/application_producer.rb

```diff
@@ -20,7 +28,7 @@ def deliver_message(msg)
         service: SERVICE,
         source: ENV['APPLICATION_TYPE']
       )
-      kafka&.deliver_message(msg.to_json, topic: self::TOPIC)
+      kafka&.produce(payload: msg.to_json, topic: self::TOPIC)&.wait
```

Comment:
JFYI: Minimum wait time is 0.1s and max 60s.
ref: https://www.rubydoc.info/github/appsignal/rdkafka-ruby/Rdkafka/AbstractHandle#wait-instance_method